pkg/csi/service/common/vsphereutil.go

```diff
@@ -934,19 +940,20 @@ func QueryVolumeSnapshotsByVolumeID(ctx context.Context, volManager cnsvolume.Ma
 }
 
 // QueryVolumeByID is the helper function to query volume by volumeID.
-func QueryVolumeByID(ctx context.Context, volManager cnsvolume.Manager, volumeID string) (*cnstypes.CnsVolume, error) {
+func QueryVolumeByID(ctx context.Context, volManager cnsvolume.Manager, volumeID string,
+	querySelection *cnstypes.CnsQuerySelection) (*cnstypes.CnsVolume, error) {
 	log := logger.GetLogger(ctx)
 	queryFilter := cnstypes.CnsQueryFilter{
 		VolumeIds: []cnstypes.CnsVolumeId{{Id: volumeID}},
 	}
-	queryResult, err := volManager.QueryVolume(ctx, queryFilter)
+	queryResult, err := utils.QueryVolumeUtil(ctx, volManager, queryFilter, querySelection, true)
 	if err != nil {
-		msg := fmt.Sprintf("QueryVolume failed for volumeID: %s with error %+v", volumeID, err)
+		msg := fmt.Sprintf("QueryVolumeUtil failed for volumeID: %s with error %+v", volumeID, err)
 		log.Error(msg)
```

Comment:
use `log.Errof` so we don't need to prepare message using `fmt.Sprintf`

Reply:
done